File: frontend/src/types/escalation.ts
export type EscalationPriority = "low" | "medium" | "high";
export type EscalationStatus = "pending" | "picked_up" | "resolved";

export interface Escalation {
  id: string;
  org_id: string;
  conversation_id: string;
  contact_name: string;
  reason: string;
  priority: EscalationPriority;
  status: EscalationStatus;
  assigned_to: string | null;
  created_at: string;
  resolved_at: string | null;
}

File: backend/app/models/escalation.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, Enum, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.enums import EscalationPriority, EscalationStatus


class Escalation(Base):
    __tablename__ = "escalations"

    org_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("organizations.id"), nullable=False, index=True
    )
    conversation_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("conversations.id"), nullable=False, index=True
    )
    reason: Mapped[str] = mapped_column(Text, nullable=False)
    priority: Mapped[EscalationPriority] = mapped_column(
        Enum(EscalationPriority, name="escalation_priority"), nullable=False
    )
    status: Mapped[EscalationStatus] = mapped_column(
        Enum(EscalationStatus, name="escalation_status"), nullable=False
    )
    assigned_to: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id"), nullable=True
    )
    resolved_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
